### PR TITLE
pam_start.c: call bindtextdomain() to expose Linux-PAM locales

### PR DESCRIPTION
**Pull request by trofi**
Without the change locales are not visible to applications using PAM if PAM library is installed into a --prefix= different from the default one.

### CODE REVIEW

**Comment by trofi** [comment]
https://github.com/linux-pam/linux-pam/issues/602 describes an issue in more detail if more context needed.

File: libpam/pam_start.c [hunk 1]
```diff
@@ -25,6 +25,21 @@ static int _pam_start_internal (
     D(("called pam_start: [%s] [%s] [%p] [%p]"
        ,service_name, user, pam_conversation, pamh));
 
+    /* Bind text domain to pull in PAM transpations for a case where
```
Comment (ldv-alt) [inline libpam/pam_start.c:28]: typo: `transpations`

File: libpam/pam_start.c [hunk 2]
```diff
@@ -25,6 +25,21 @@ static int _pam_start_internal (
...
+       has a chance to have some overhead. Let's try to do it once (or a
+       small number of times as `bound_text_domain` is not protected by
+       a lock. */
+    static int bound_text_domain = 0;
+    if (!bound_text_domain) {
+	bound_text_domain = 1;
+#ifdef HAVE_BINDTEXTDOMAIN
+	bindtextdomain(PACKAGE, LOCALEDIR);
+#endif
+    }
```
Comment (ldv-alt) [inline libpam/pam_start.c:41]: Let's move the whole hunk under `#ifdef HAVE_BINDTEXTDOMAIN`.

**Review by ldv-alt — CHANGES_REQUESTED** [review]

File: libpam/pam_start.c (hunk 1 above)
Comment (trofi) [reply libpam/pam_start.c]: Fixed typo.

File: libpam/pam_start.c (hunk 2 above)
Comment (trofi) [reply libpam/pam_start.c]: Moved all under `#ifdef HAVE_BINDTEXTDOMAIN`

**Review by ldv-alt — APPROVED** [review]

**Pull request merged**